roadmap/implementers-guide/src/runtime/paras.md

```diff
@@ -49,15 +53,69 @@ struct ParaGenesisArgs {
   /// True if parachain, false if parathread.
   parachain: bool,
 }
+
+/// The possible states of a para, to take into account delayed lifecycle changes.
+pub enum ParaLifecycle {
+  /// Para is new and is onboarding as a Parathread.
+  OnboardingAsParathread,
```

Comment:
is it really worth complicating things by enabling onboarding directly to parachain? i expect that 99% of the time, teams will onboard to parathread, upgrade to parachain, downgrade to parathread and deregister the parathread. i very much doubt there will be any need (or much desire) for `OnboardingAsParachain` or `OutgoingParathread` states.

Reply:
This pallet has no such restrictions with the onboarding story. Those limitations will be imposed by the Registrar pallet.

Reply:
@gavofyork It'd be helpful for testing / Rococo where we onboard teams as parachains. Otherwise, I'd request that this PR come with a change to the `ProposeParachain` pallet to onboard proposals as parathreads first.

Reply:
> This pallet has no such restrictions with the onboarding story. Those limitations will be imposed by the Registrar pallet.

Yeah, but the extra two enum items and the logic that is needed to handle them complicate things, and I don't think they're going to be needed in reality (except, apparently, for testing & Rococo).

Reply:
I could also imagine direct parachain registration outside of the registrar module for system level parachains. Having the API handle that kind of scneario through root calls for example sounds relevant to me...

Please tell me if you feel strongly here, I will happily oblige, but I have made changes in this pallet independent of any parachain onboarding story we have. More as a general API backend.

Reply:
I have no burning desire to changes things *right now*, however when we eventually add a new kind of para (or two), this enum will quickly become unwieldy if the model of switching directly between any kind of para (or no para) persists, so I view a parathread-centric transition model to be the only way forward long-term.

Reply:
@gavofyork That's reasonable, especially as you mention other ways that parachains might share cores in the future. However, anything we do there would need to come with significant changes in the `Scheduler` module.